grpc/client.go

```diff
@@ -44,13 +46,36 @@ func NewWithDialler(address string, d Dialler) (Connector, error) {
 	}
 
 	return clientImpl{
-		addr: address,
-		d:    d,
+		addr:    address,
+		d:       d,
+		tlsfile: tls,
 	}, nil
 }
 
 // NewConn creates a new gRPC connection
 func (c clientImpl) NewConn() (grpc.ClientConnInterface, Closer, error) {
+	log.Println("files", c.tlsfile)
+	if c.tlsfile != "" {
+		config, err := ParseTLSFiles(c.tlsfile)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to parse TLS files: %w", err)
```

Comment:
Does a client need to have access to the underlying error (%w)?

Reply:
client is responsible for providing the certificate, then i believe it is ok to provide the underlying error .

Reply:
Usually %w is used in order a client could extract the original error here and perform recovery based on that error